### PR TITLE
feat(folder-structure): add option to use default folder/file structure

### LINKED ISSUE #17: Create basic folder structure
Add folders and files via `setup.sh`

```sh
app/jade/index.jade
app/scripts/main.js
app/styles/main.stylus
```

### PR DESCRIPTION
**Pull request by mrMetalWood**
closes #17

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -74,3 +74,27 @@ if [[ ${PROJECT_TYPE} == 'nodejs' ]]; then
...
+
+if [[ ${DEFAULT_FOLDER_STRUCTURE} == 'y' ]]; then
+
+  mkdir -p app/{jade/layouts,styles,scripts}
+
+  echo "Created folder structure:"
+  find app -type d -print
+
+  fetch "jade-default"
+  mv jade-default app/jade/layouts/_default.jade
```
Comment (yfr) [inline setup.sh]: Please use files names without underscores. This concept is old. Per default we compile targets from teh root folder and the layout should be private at all circumstances.

File: setup.sh [hunk 2]
```diff
@@ -74,3 +74,27 @@ if [[ ${PROJECT_TYPE} == 'nodejs' ]]; then
...
+read -p "Use default folder structure? [y, N]: " DEFAULT_FOLDER_STRUCTURE </dev/tty
+
+if [[ ${DEFAULT_FOLDER_STRUCTURE} == 'y' ]]; then
+
+  mkdir -p app/{jade/layouts,styles,scripts}
+
+  echo "Created folder structure:"
+  find app -type d -print
+
+  fetch "jade-default"
```
Comment (yfr) [inline setup.sh]: Can we have a second parameter to the `fetch()` function with the target? Would save one step.

File: setup.sh [hunk 3]
```diff
@@ -74,3 +74,27 @@ if [[ ${PROJECT_TYPE} == 'nodejs' ]]; then
...
 fi
+
+read -p "Use default folder structure? [y, N]: " DEFAULT_FOLDER_STRUCTURE </dev/tty
+
+if [[ ${DEFAULT_FOLDER_STRUCTURE} == 'y' ]]; then
+
+  mkdir -p app/{jade/layouts,styles,scripts}
+
+  echo "Created folder structure:"
+  find app -type d -print
```
Comment (yfr) [inline setup.sh]: Nice!

File: templates/jade-default [hunk 4]
```diff
@@ -0,0 +1,19 @@
+doctype html
+html(lang="en")
+  head
+    title
```
Comment (yfr) [inline templates/jade-default]: Can we add the project title here?

```jade
title {{project-name}}

File: templates/jade-index [hunk 5]
```diff
@@ -0,0 +1,5 @@
+extends layouts/_default
+
+block content
+
+  h1 Ubilabs Project Template
```
Comment (yfr) [inline templates/jade-index]: Maybe add the description of the project.

```jade
p {{project-description}}
```

Comment (yfr) [inline templates/jade-index]: Also maybe the project title in the `h1`?

**Comment by yfr** [comment]
Nice one! Just some ideas. maybe not in this PR. But would be nice i think. Also maybe we should implement a generic "sed" function to replace project properties along the created files.

**Comment by mrMetalWood** [comment]
Added everything except the generic `sed` function. This is probably better off in a sperate PR.
:trophy:

**Pull request merged**